src/datastore/database.hpp

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include <vector>
+#include "souvenir.hpp"
+#include "stadium.hpp"
+#include "team.hpp"
+
+/**
+ * @class Database class
+ *
+ * This class holds all of the programs data.
+ * It contains a map of teams and a map of stadiums.
+ * Each stadium has a vector of souvenirs within it's class.
+ *
+ * All the members aside from getEnumValue are static
+ */
+class Database
+{
+public:
+    /* Static methods to Load and Save from/to file */
+    static void loadFromFile(const std::string&);
+    static void saveToFile(const std::string&);
+
+     /* Getters */
+    static std::map<int,Team> getTeams();
+    static std::map<int,Stadium> getStadiums();
```

Comment:
Return a const-ref

Reply:
I think it should be by ref so the admin view can edit it

Reply:
But why would you ever need to edit the map? They would edit the items within the map, but not the map itself

Reply:
Clarification: they would edit COPIES of the items within the map

Reply:
Yeah, but that returns the map by value, so the actual map is not edited. Also, he converts the maps to the vectors to write out the data, so the map's internals need to be edited.

Reply:
Oh, that is done by the find functions (I believe). The find functions returns a reference of teams and stadiums

Reply:
Nope... it's also a const-ref :pepeHmm: